src/mjolnir/shortcutbuilder.cc

```diff
@@ -386,10 +386,7 @@ uint32_t AddShortcutEdges(GraphReader& reader,
   for (uint32_t i = 0; i < edge_count; i++, ++edge_id) {
     // Skip transit connection edges.
     const DirectedEdge* directededge = tile->directededge(edge_id);
-    if (directededge->use() == Use::kTransitConnection ||
-        directededge->use() == Use::kEgressConnection ||
-        directededge->use() == Use::kPlatformConnection || directededge->bss_connection() ||
-        directededge->use() == Use::kConstruction) {
+    if (!directededge->can_form_shortcut()) {
```

Comment:
there is another similar piece of code above that i think can make use of this function, its up on line 200

Reply:
also in that method `CanContract` there are a bunch of stipulations that the node has to follow as well, it would be nice to add that as a function to nodeinfo and then maybe we need to use it in `GetShortcut` and `RecoverShortcut` as well?

Reply:
definitely, i noticed that too and hoped you would as well :smile: